lib/evmone/baseline.cpp

```diff
@@ -356,6 +373,20 @@ evmc_result execute(const VM& vm, ExecutionState& state, const CodeAnalysis& ana
 evmc_result execute(evmc_vm* c_vm, const evmc_host_interface* host, evmc_host_context* ctx,
     evmc_revision rev, const evmc_message* msg, const uint8_t* code, size_t code_size) noexcept
 {
+    if (rev >= EVMC_CANCUN && is_eof_container({code, code_size}))
+    {
+        // TODO(EOF): The initcode must be validated. Doing this just before execution is
+        //   good because validation can be combined with analysis/loading. But consider also
+        //   other places like create instructions.
+        if (const auto err = validate_eof(rev, {code, code_size});
```

Comment:
Hm, if we have to validate prior to execution, don't we lose any of the potential speed ups?

Shouldn't we have an EVMC option for this perhaps (force validate execution)?

Reply:
@chfast ^ Can you please elaborate on this.

Reply:
We should remove it from this PR and add later in different form. I created the issue about it: https://github.com/ethereum/evmone/issues/585.

Reply:
Removed.